src/oscilloscope.cpp

```diff
@@ -3238,14 +3238,17 @@ void Oscilloscope::editMathChannelFunction(int id, const std::string& new_functi
 	int current_x = index_x;
 	int current_y = index_y;
 
-	triggerRightMenuToggle(
-		static_cast<CustomPushButton* >(ui->btnAddMath), false);
-	triggerRightMenuToggle(
-		static_cast<CustomPushButton* >(chn_widget->menuButton()), true);
-	math_pair->first.btnAddChannel->setText("Add Channel");
-	math_pair->second->setFunction("");
-	addChannel = true;
-	ch_ui->btnEditMath->setChecked(false);
+
+	if (new_function != chn_widget->function().toStdString()) {
```

Comment:
Wouldn't it make sens to check this at the beginning of the function? And in case the new function is the same as the current on, just return?

Reply:
That's the way it was done. However, when the probe_attenuation changes (aprox line 679), we need to update the rail block based on the new values (which sets a max and a min for math output values). If this function is not called, the rail block will not update, and the results will not look good.  

I might be able to update the rail block on the run, without needing to disconnect and reconnect all the blocks. Let me look into it.

Reply:
Added the check back in the function. We can update the rail max and min values at runtime, so there is no need to edit the math channel every time.